src/puppetlabs/puppetdb/cheshire.clj

```diff
@@ -26,19 +26,27 @@
 (defn add-common-json-encoders!*
   "Non-memoize version of add-common-json-encoders!"
   []
-  (generate/add-encoder
+  (add-encoder
     org.postgresql.util.PGobject
     (fn [^PGobject data ^JsonGenerator jsonGenerator]
       ;; The .getPrettyPrinter method on the Jackson jsonGenerator will return
       ;; nil if `:pretty` is not set as a cheshire option
       (if (.getPrettyPrinter jsonGenerator)
-        (generate/encode-map (core/parse-string (.getValue data)) jsonGenerator)
+        (let [obj (core/parse-string (.getValue data))
+              encode-fn (condp instance? obj
+                          clojure.lang.IPersistentMap encode-map
+                          clojure.lang.IPersistentVector encode-seq
+                          clojure.lang.IPersistentSet encode-seq
```

Comment:
It's not possible for this to come back from the DB is it? I would think that's true of most of the things in this list.